pkg/certs/cert-inspection/certgraphanalysis/deduplication.go

```diff
@@ -1,34 +1,68 @@
 package certgraphanalysis
 
-import "github.com/openshift/library-go/pkg/certs/cert-inspection/certgraphapi"
+import (
+	"github.com/openshift/library-go/pkg/certs/cert-inspection/certgraphapi"
+)
 
 func deduplicateCertKeyPairs(in []*certgraphapi.CertKeyPair) []*certgraphapi.CertKeyPair {
 	ret := []*certgraphapi.CertKeyPair{}
 	for _, currIn := range in {
-		found := false
+		if currIn == nil {
+			panic("currIn is nil")
+		}
+
+		foundIdx := -1
 		for j, currOut := range ret {
-			if currIn == nil {
-				panic("one")
-			}
 			if currOut == nil {
-				panic("two")
+				panic("currOut is nil")
 			}
-			if currOut.Name == currIn.Name {
-				ret[j] = combineSecretLocations(ret[j], currIn.Spec.SecretLocations)
-				ret[j] = combineCertOnDiskLocations(ret[j], currIn.Spec.OnDiskLocations)
-				found = true
-				break
+			// currOut has no name set - skip and allow it to be found later
+			if len(currOut.Name) == 0 {
+				continue
+			}
+			switch len(currIn.Name) {
+			case 0:
+				// Certificate and key have matching modulus
+				if currOut.Spec.CertMetadata.CertIdentifier.PubkeyModulus == currIn.Spec.CertMetadata.CertIdentifier.PubkeyModulus {
+					foundIdx = j
+				}
+			default:
+				// Two certificates with matching names
+				if currOut.Name == currIn.Name {
+					foundIdx = j
+				}
+			}
+			// Append currIn locations to found certkeypair
+			if foundIdx != -1 {
+				ret[foundIdx] = combineSecretLocations(ret[foundIdx], currIn.Spec.SecretLocations)
+				ret[foundIdx] = combineCertOnDiskLocations(ret[foundIdx], currIn.Spec.OnDiskLocations)
```

Comment:
I know this is pre-existing, but it looks like a mistake for our purposes where knowing what is reflected on both cluster and and ondisk only is valuable.